atomic_reactor/plugins/prepub_squash.py

```diff
@@ -55,20 +56,28 @@ class PrePublishSquashPlugin(PrePublishPlugin):
     # Fail the build in case of squashing error
     can_fail = False
 
-    def __init__(self, tasker, workflow, tag=None, from_layer=None, remove_former_image=True,
-                 dont_load=False):
+    def __init__(self, tasker, workflow, tag=None, from_base=False, from_layer=None,
+                 remove_former_image=True, dont_load=False):
         """
         :param tasker: DockerTasker instance
         :param workflow: DockerBuildWorkflow instance
-        :param from_layer: The layer from we will squash - by default it'll be the first layer
+        :param from_base: bool, squash from base-image layer, can't be specified with from_layer
+        :param from_layer: layer from we will squash - by default it'll be the first layer
         :param tag: str, new name of the image - by default use the former one
         :param remove_former_image: bool, remove unsquashed image?
         :param dont_load: bool, don't load squashed image into Docker, place it to `$tmpdir/image.tar` instead
         """
         super(PrePublishSquashPlugin, self).__init__(tasker, workflow)
         self.image = self.workflow.builder.image_id
-        self.from_layer = from_layer
         self.tag = tag or str(self.workflow.builder.image)
+        if from_base and from_layer is not None:
+            raise RuntimeError("Both: 'from_base' and 'from_layer' specified.")
+        self.from_layer = from_layer
+        if from_base:
+            df_path, _ = self.workflow.source.get_dockerfile_path()
```

Comment:
this won't work, because squash tool expects ID

I think we should use [this attribute](https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/inner.py#L297) of `inner.py` so we don't inspect images all over the place.